### ISSUE TITLE
No error if test library has multiple functions resulting same keyword

### ISSUE THREAD

**Issue by pekkaklarck** [robotframework/robotframework#2097]
For example, the following library would only create one keyword `Example Keyword`. Which implementation would be used depends on the order functions are processed:

``` python
def example_keyword():
    print 'first'

def exampleKeyword():
    print 'second'

def examplekeyword():
    print 'third'

def ExampleKeyword():
    print 'fourth'
```

With user keywords you would get an error that there are multiple implementations and trying to use the keyword would fail too. That's probably a good approach also with libraries.

Same problem occurs also when specifying the keyword name explicitly and when the explicit name contains embedded arguments. See #2095 about somewhat related regression with user keywords.

Because this is a pretty low level bug and there is a potential backwards incompatibility risk, I think it is best to wait until RF 3.0 before fixing this.

**Comment by pekkaklarck** [comment]
It turned out that the fix to #2095 changed the behavior also with duplicate library keywords. At the moment these errors aren't handled correctly, but it is a better idea to fix error reporting than trying to avoid the error.

**Comment by pekkaklarck** [comment]
After the changes, creating same keyword multiple times, like in the original example, causes an error and trying to use the keyword fails.
